src/components/hmi_message_handler/src/mqueue_adapter.cc

```diff
@@ -48,14 +49,22 @@ class ReceiverThreadDelegate : public threads::ThreadDelegate {
   ReceiverThreadDelegate(mqd_t mqueue_descriptor,
                          HMIMessageHandler* hmi_message_handler)
       : mqueue_descriptor_(mqueue_descriptor)
-      , hmi_message_handler_(hmi_message_handler) {}
+      , hmi_message_handler_(hmi_message_handler)
+      , shutdown_requested_(false) {}
 
  private:
-  virtual void threadMain() {
-    while (true) {
+  void threadMain() OVERRIDE {
+    while (!shutdown_requested_) {
       static char buffer[kMqueueMessageSize];
-      const ssize_t size =
-          mq_receive(mqueue_descriptor_, buffer, kMqueueMessageSize, NULL);
+
+      struct timespec current_time;
```

Comment:
@vlantonov The `<time.h>` header declares the structure timespec and functions below. And looks that it's missed